<a href="https://colab.research.google.com/github/yunjung-lee/python_mini_project_skin_cancer/blob/master/skin_cancer.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##########  skin cancer in kaggle dataset

#### used knn ,dropout,save file

In [0]:
!git clone https://github.com/yunjung-lee/python_mini_project_skin_cancer.git

!git clone  :  깃을 접속함
https://github.com/yunjung-lee/python_mini_project_skin_cancer.git : 깃의 주소


In [0]:
import numpy as np
import tensorflow as tf
from sklearn.model_selection import train_test_split
import sklearn.preprocessing

opne data(data name :  hmnist_28_28_RGB.csv)

In [0]:
data = np.loadtxt('drive/Colab Notebooks/dataset/hmnist_28_28_RGB.csv', dtype = np.float32, delimiter = ",", skiprows = 1,encoding = "utf-8")
# print(data)
xdata = data[:,:-1]
ydata = data[:,[-1]]



one_hot  encoding

In [0]:
one_hot = sklearn.preprocessing.LabelBinarizer()
ydata = one_hot.fit_transform(ydata)

train_ test_split

In [0]:
X_train, X_test, y_train, y_test = train_test_split(xdata, ydata, test_size=0.33, random_state=42)

nn-learning&drop-out 선언부
output에서 sigmoid 사용에  softmax_cross_entropy_with_logits를 사용

In [0]:
x = tf.placeholder(tf.float32, [None,28*28*3])
y = tf.placeholder(tf.float32, [None,7])
keep_prob=tf.placeholder(tf.float32)

w1 = tf.Variable(tf.random_normal([28*28*3,256],stddev=0.01))
L1 = tf.nn.relu(tf.matmul(x,w1))
L1=tf.nn.dropout(L1,keep_prob)

w2 = tf.Variable(tf.random_normal([256,256],stddev=0.01))
L2 = tf.nn.relu(tf.matmul(L1,w2))
L2=tf.nn.dropout(L2,keep_prob)

w3 = tf.Variable(tf.random_normal([256,7],stddev=0.01))
model = tf.matmul(L2,w3)

cost = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(logits=model,labels=y))
optimizer = tf.train.AdamOptimizer(0.001).minimize(cost)


신경망 모델 학습

 batch size를 100으로 두고 drop-out을 0.8로 설정하여 진행

In [16]:
init = tf.global_variables_initializer()
saver = tf.train.Saver()
sess = tf.Session()
sess.run(init)

batch_size = 100
total_batch = int(len(X_train[0])/batch_size)
for epoch in range(20):
    total_cost = 0

    for i in range(total_batch):
        _, cv = sess.run([optimizer,cost],feed_dict={x:X_train,y:y_train,keep_prob:0.8})
        total_cost += cv
#     print("epoch :", " %d" % (epoch+1), "avg cost : ","{:.3f}".format(total_cost/total_batch))
    is_correct = tf.equal(tf.argmax(model,1), tf.argmax(y,1))
    accuracy = tf.reduce_mean(tf.cast(is_correct, dtype=tf.float32))
print("정확도 : ", sess.run(accuracy,feed_dict={x:X_test,y:y_test,keep_prob:1.0}))

saver.save(sess, 'drive/Colab Notebooks/datasetcnn_session')
sess.close()


정확도 :  0.66596067


결과 avg cost :  2.989
정확도 :  0.66596067
epoch :  2 avg cost :  1.084
정확도 :  0.66596067
epoch :  3 avg cost :  1.006
정확도 :  0.66596067
epoch :  4 avg cost :  0.992
정확도 :  0.66596067
epoch :  5 avg cost :  0.959
정확도 :  0.66596067
epoch :  6 avg cost :  0.940
정확도 :  0.66596067
epoch :  7 avg cost :  0.927
정확도 :  0.66596067
epoch :  8 avg cost :  0.921
정확도 :  0.66596067
epoch :  9 avg cost :  1.014
정확도 :  0.66596067
epoch :  10 avg cost :  0.955
정확도 :  0.66596067
epoch :  11 avg cost :  0.938
정확도 :  0.66596067
epoch :  12 avg cost :  0.935
정확도 :  0.66596067
epoch :  13 avg cost :  0.921
정확도 :  0.66596067
epoch :  14 avg cost :  0.921
정확도 :  0.66596067
epoch :  15 avg cost :  0.920
정확도 :  0.66596067
epoch :  16 avg cost :  0.952
정확도 :  0.66596067
epoch :  17 avg cost :  0.916
정확도 :  0.66656584
epoch :  18 avg cost :  0.911
정확도 :  0.66807866
epoch :  19 avg cost :  0.917
정확도 :  0.66717094
epoch :  20 avg cost :  0.903
정확도 :  0.66747355